### PR TITLE
MM-60436 - remove telemetry api track calls

### PR DESCRIPTION
**Pull request by catalintomai**
#### Summary
Removing the "api" category telemetry calls, with the following exceptions:
- `api_posts_create` , but only when "Message Priority" feature is used
- `api_posts_ack` - needed for "Message Acknowledgement" feature usage
- `api_groups_get_associated_to_channel` - needed for "Custom User Groups" feature usage
- `api_teams_invite_guests` - needed for "Guest Accounts" feature usage

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60435
https://mattermost.atlassian.net/browse/MM-60436

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/platform/client/src/client4.ts [hunk 1]
```diff
@@ -2183,19 +2070,12 @@ export default class Client4 {
             analyticsData.priority = post.metadata.priority.priority;
             analyticsData.requested_ack = post.metadata.priority.requested_ack;
             analyticsData.persistent_notifications = post.metadata.priority.persistent_notifications;
-        }
-
-        this.trackEvent('api', 'api_posts_create', analyticsData);
-
-        if (result.root_id != null && result.root_id !== '') {
-            this.trackEvent('api', 'api_posts_replied', analyticsData);
+            this.trackEvent('api', 'api_posts_create', analyticsData);
```
[the remark this answers is not in the document]
Comment (catalintomai) [reply webapp/platform/client/src/client4.ts:2073]: yes, that was the purpose - to collect `api_posts_create` only when that feature is on.

**Review by lieut-data — APPROVED** [review]
:+1: to reducing our dependence on this style of telemetry. Do we have a plan to migrate the remaining use cases into the daily ping instead of per-API?

**Comment by catalintomai** [comment]
> 👍 to reducing our dependence on this style of telemetry. Do we have a plan to migrate the remaining use cases into the daily ping instead of per-API?

No plan fwik - first step was to remove the high-volume unused ones. Re: the few remaining ones (which given the estimated usage - see https://mattermost.looker.com/dashboards/438 - they are quite low-volume) , I think we need to rationalize/re-think the need to capture usage for end-user paid (right now surfaces in the company scorecard) and new features + the best/right way to capture them (could be client/api events, could be server/daily ping events, could be smth else).

**Review by ifoukarakis — APPROVED** [review]

**Review by jwilander — APPROVED** [review]

**Comment by lieut-data** [comment]
> I think we need to rationalize/re-think the need to capture usage for end-user paid (they are right now surfaced in the company scorecard) and new features + the best/right way to capture them (could be client/api events, could be server/daily ping events, could be smth else).

I'm aligned with moving away from per-user events towards exclusively daily pings. Any concerns with pushing towards that as an outcome unless we see evidence to head elsewhere, @catalintomai?

**Comment by catalintomai** [comment]
> per-user events
@lieut-data , one possible concern, and that is why it might be a good idea to have a larger conv. before deciding on the switch is possible loss of signal if we move away from per-user events (not that we are using the per-user info in a significant manner currently), due to the move towards per-user/premium SKU (BillA seems to know more about the impending packaging changes). If we will need to capture per-user telemetry to distinguish usage per SKU we might need to keep the existing granularity (not that the events could not be collected/aggregated at the server, per-user).

**Comment by lieut-data** [comment]
> ... due to the move towards per-user/premium SKU ...

Thanks, @catalintomai! Indeed, there's been lots of discussion around that recently, but just this week heard that we're aligned to staying on the "all-you-can-eat" model for licenses vs. trying to do per-user. So at least on this front, it shouldn't be a blocker for telemetry.

**Comment by catalintomai** [comment]
Thanks for the update @lieut-data. Given that there just was an extensive discussion on the subject in https://hub.mattermost.com/private-core/pl/h67ztapto3bs7ya8k9qs3eqzrr, wasn't aware that the dust settled on the issue. 
Given that it is a departure from current, I'd still favor documenting the decision - here: https://mattermost.atlassian.net/wiki/spaces/DATAENG/overview?homepageId=2403532962 or in some other place + reasons behind it (like is it because we want to be aligned with off-line data gathering, Rudder cost, maintenance cost, lack of usage, other reasons ?) + pros/cons, so that it can be more easily shared + in the future, if/when someone would propose to go back to per-user events, there is some info/trace regarding the reasoning behind it.

**Comment by lieut-data** [comment]
Absolutely, @catalintomai -- very hot off the press, as it were, and partly influenced by exactly that conversation and the complexities it creates. I'm sure we'll hear more directly from Chen & Bill soon on this front.

**Review by sbishel — APPROVED** [review]

**Pull request merged**